common/src/node/ethereum/provider.rs

```diff
@@ -0,0 +1,47 @@
+use crate::EthError;
+use ethers::providers::{Http, Provider};
+use url::Url;
+
+#[cfg(not(target_arch = "wasm32"))]
+use once_cell::sync::OnceCell;
+#[cfg(not(target_arch = "wasm32"))]
+static G_AGENTINFO: OnceCell<String> = OnceCell::new();
+
+#[cfg(not(target_arch = "wasm32"))]
+pub fn set_ethers_httpagent(agent: &str) -> Result<(), EthError> {
+    if G_AGENTINFO.get().is_none() {
+        G_AGENTINFO.set(agent.to_string()).expect("set agent info");
```

Comment:
return an error instead of a panic? or no need for `set_ethers_httpagent` (just set via the environment variable?)?

Reply:
set_ethers_httpagent is necessary for ios, android
i'll check returning an error